### PR TITLE
Fix/fee history base fee

### PR DESCRIPTION
**Pull request by Eikix**
Time spent on this PR:

Resolves: #827 

# Pull Request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# What is the new behavior?

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by Eikix** [comment]
Also fixes @bajpai244 's problem on docker compose as boyscout commit

File: src/eth_provider/provider.rs [hunk 1]
```diff
@@ -417,34 +418,31 @@ where
...
         }
 
-        let end_block = U256::from(self.tag_into_block_number(newest_block).await?);
-        let end_block_plus = end_block + U256::from(1);
+        let end_block = self.tag_into_block_number(newest_block).await?;
+        let end_block_plus = end_block + U64::from(1_u64);
 
         // Clamp the block count to the range [0, end_block_plus]
-        block_count = Ord::clamp(block_count, U256::ZERO, end_block_plus);
+        let block_count = Ord::clamp(U64::from(block_count.to()), U64::ZERO, end_block_plus);
```
Comment (greged93) [inline src/eth_provider/provider.rs:433]: Should we have all variables be `u64`? It looks like we convert a lot to U64 but could probably just be u64 no?

File: src/eth_provider/provider.rs [hunk 2]
```diff
@@ -458,10 +456,14 @@ where
             })
             .collect::<Vec<_>>();
 
+        let mut base_fee_per_gas =
+            blocks.iter().map(|header| header.header.base_fee_per_gas.unwrap_or_default()).collect::<Vec<_>>();
+        base_fee_per_gas.extend_from_within((base_fee_per_gas.len() - 1)..);
```
Comment (greged93) [inline src/eth_provider/provider.rs:461]: so we take the last element of the vec and append it to the vec? Is that the spec?

Comment (Eikix) [reply src/eth_provider/provider.rs:461]: no no the spec is to check if gas ratio > 50% then bump base fee per gas by 12.5% probably

File: src/eth_provider/provider.rs (hunk 1 above)
Comment (greged93) [reply src/eth_provider/provider.rs:433]: I just mean there is a lot of conversion between u64 and U64 in this function, maybe we can just use `u64` everywhere?

File: src/eth_provider/provider.rs (hunk 2 above)
Comment (greged93) [reply src/eth_provider/provider.rs:461]: ah ok ok but for now you just append the last one?

File: src/eth_provider/provider.rs [hunk 3]
```diff
@@ -427,10 +427,11 @@ where
         }
 
         let end_block = self.tag_into_block_number(newest_block).await?;
-        let end_block_plus = end_block + U64::from(1_u64);
+        let end_block = end_block.to::<u64>();
+        let end_block_plus = end_block + 1;
```
Comment (greged93) [inline src/eth_provider/provider.rs:431]: nit: one of the reviewers on Reth suggested to use `saturating_add` and `saturating_sub`.

**Review by greged93 — APPROVED** [review]

**Pull request merged**